### ISSUE TITLE
Some colors aren't displayed correctly when spelled out.

### ISSUE THREAD

**Issue by adriandeveloper** [ap/vim-css-color#56]
There are some colors that aren't displayed correctly when you type them out. For example. Type in purple and it's highlighted in gray, but if you input it's hex equivalent then you get the right color.

**Comment by ap** [comment]
I cannot reproduce this. If you move your cursor onto an incorrectly highlighted word and give this command,

````vim
:echo map(synstack(line('.'), col('.')), 'synIDattr(v:val, "name")')
````

… what is the output?

Which other color keywords does this affect? “Some” isn’t much help… :smile:

**Comment by adriandeveloper** [comment]
Yeah it wasn't helpful was it? lol. Sorry, was in a rush.  

One of the colors I've used is purple.  It looks like the color isn't shown but when you enter the hex value of it, the color pops up.    In regards to the output I get:

['cssDefinition', 'cssAttrRegion', 'BG800080']

Attached is a screenshot.


![screenshot 2015-09-30 05 50 55](https://cloud.githubusercontent.com/assets/3031275/10191180/59e787d4-6737-11e5-9cdd-83df398bff95.png)

**Comment by ap** [comment]
OK, so, that bit looks right. I guess we’ll have to play 20 questions; sorry for the bother.

1. What does `:hi BG800080` say in that situation?
2. If you type `#800080` instead of `purple`, what colour do you get?

**Comment by adriandeveloper** [comment]
Np.  

Answer to question 1:

![screenshot 2015-09-30 17 09 30](https://cloud.githubusercontent.com/assets/3031275/10208090/5bba7184-6796-11e5-971e-451c86e13e07.png)

Answer to questions 2 & 3:

![screenshot 2015-09-30 17 11 14](https://cloud.githubusercontent.com/assets/3031275/10208095/6af36412-6796-11e5-9df1-540d0762dca9.png)


Thing is after I typed #BG800080 then it worked as per the screenshot. 

Now when I quit vim and made up a new .css stylesheet and typed the color in again, it didn't work again.

**Comment by ap** [comment]
\*headscratch*

Is the output of `:hi BG800080` different after you’ve typed `#800080` and `purple` gets highlighted correctly?

**Comment by ap** [comment]
Yes there is. No need to show me, I found it. This is painfully stupid.

It’s a late effect of #16, itself long fixed in commit c8ae081cf7. Once upon a time, the mapping of RGB values to 256-colour terminal palette index was recalculated at every Vim startup. Then, at some point, to make startup faster, I dumped the results of the conversion and hardcoded them in the script. But I did this back when the mapper function was simply and stupidly and utterly broken. As a result I hardcoded a bunch of broken outputs into the script. Arrrrrrgghhh.
